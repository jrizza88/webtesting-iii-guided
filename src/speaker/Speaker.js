import React from 'react';

const Speaker = props => {
    return (
        <>
            <button onClick={props.speak}>Speak</button>
            <div>{props.message}</div>
            <button onClick={props.save}>Save</button>
            <button onClick={props.delete}>Delete</button>
        </>
    )
}

export default Speaker;