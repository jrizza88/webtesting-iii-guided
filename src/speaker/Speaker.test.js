import React from 'react';
import {render, fireEvent } from 'react-testing-library';
import renderer from 'react-test-renderer';
import Speaker from './Speaker';

describe('<Speaker />', () => {
    it('should match the snapshot with a message', () => {
        const tree = renderer.create(<Speaker />)

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it.skip('second snapshot message', () => {
        const tree = renderer.create(<Speaker />)

        
    });

    it('should call the speak function on click ', () => {

        const speakMock = jest.fn();

            // get by text
const { getByText } = render(<Speaker speak={speakMock}/>)

const speakButton = getByText(/speak/i)

fireEvent.click(speakButton);

expect(speakMock).toHaveBeenCalled()
    });
});