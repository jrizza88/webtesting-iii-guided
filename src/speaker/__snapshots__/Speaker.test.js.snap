// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Speaker /> should match the snapshot with a message 1`] = `
Array [
  <button>
    Speak
  </button>,
  <div />,
  <button>
    Save
  </button>,
  <button>
    Delete
  </button>,
]
`;
